### PR TITLE
Gnm more changes

### PR DESCRIPTION
**Pull request by kain88-de**
Changes made in this Pull Request:
 - another style change  in the neighbor_search (removed explicit loops)
 - speed up tests

PR Checklist
------------
 - [x] Tests?
 - ~[ ] Docs?~
 - ~[ ] CHANGELOG updated?~
 - ~[ ] Issue raised/referenced?~

### CODE REVIEW

File: package/MDAnalysis/analysis/gnm.py [hunk 1]
```diff
@@ -164,20 +163,18 @@ def neighbour_generator(positions, cutoff):
...
     n_z = len(grid[0][0])
-    for cell_x, cell_y, cell_z in itertools.product(range(n_x), range(n_y), range(n_z)):
+    for cell_x, cell_y, cell_z in itertools.product(
+            range(n_x), range(n_y), range(n_z)):
         atoms = grid[cell_x][cell_y][cell_z]
         # collect all atoms in own cell and neighboring cell
         all_atoms = []
-        for x in (-1, 0, 1):
+        nei_cells = (-1, 0, 1)
+        for x, y, z in itertools.product(nei_cells, nei_cells, nei_cells):
```
Comment (kain88-de) [inline package/MDAnalysis/analysis/gnm.py]: @richardjgowers I forgot to use the product generator here

File: testsuite/MDAnalysisTests/analysis/test_gnm.py [hunk 2]
```diff
@@ -37,119 +37,115 @@
...
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
 
 
 def test_closeContactGNMAnalysis(universe):
-    gnm = MDAnalysis.analysis.gnm.closeContactGNMAnalysis(universe, weights="size")
-
-    gnm.run()
-
+    gnm = mda.analysis.gnm.closeContactGNMAnalysis(universe, weights="size")
+    gnm.run(stop=2)
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_gnm.py]: this is the speed up secret. Iterating over the whole trajectory just takes enormous amounts of time. Reducing to just looking if the first two frames are OK speeds things up significantly.

File: testsuite/MDAnalysisTests/analysis/test_gnm.py [hunk 3]
```diff
@@ -37,119 +37,115 @@
...
     (False, None),
     (True, 'size')
 ))
-def test_closeContactGNMAnalysis(universe, MassWeight, expected_value):
-    gnm = MDAnalysis.analysis.gnm.closeContactGNMAnalysis(universe,
-                                                          MassWeight=MassWeight)
-    assert gnm.weights == expected_value
+def test_closeContactGNMAnalysis_weight_dep(universe, MassWeight,
+                                            expected_value):
+    with pytest.deprecated_call():
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_gnm.py]: I like to check that we corretly raise deprecation warnings.

**Comment by kain88-de** [comment]
Awesome the tests are now way way faster on Travis. This likely cuts more then a minute from the test times.

File: package/MDAnalysis/analysis/gnm.py [hunk 4]
```diff
@@ -245,8 +246,15 @@ def __init__(self, universe, selection='protein and name CA', cutoff=7.0,
                             if Bonus_groups else []
         self.ca = self.u.select_atoms(self.selection)
 
-    def _generate_output(self, w, v, outputobject, time, matrix, nmodes=2,
-                         ReportVector=None, counter=0):
+    def _generate_output(self,
```
Comment (richardjgowers) [inline package/MDAnalysis/analysis/gnm.py]: this looks pretty ugly imo

Comment (kain88-de) [inline package/MDAnalysis/analysis/gnm.py]: You want me to change it back?

**Pull request merged**